### DISCUSSION TITLE
Volume Stream for Reactive Lights??

### DISCUSSION THREAD

**Discussion by eroboticdude** [pschatzmann/arduino-espeak-ng#10] [Q&A] [answered]
Hello, I am working on a project that involves a voice reading text downloaded from wikipedia. Everything works, but I want to add reactive lighting with FastLED. How would I go about setting this up? The program pauses everything when a speech call is made and does not allow for any other processes during speech. The esp32-s2 is single core, so for now I am limited to that. In the future it will be easier to use the second core, but now I need it to work on the esp I have. Should I use a Free Rtos function for the speech call? I don't understand enough about this library for me to easily figure out how to do that. I have not used callbacks, but I'm guessing that's a good place to start.

**Comment by pschatzmann** [comment, marked as answer]
I suggest to give it a try. There is a chaper in the wiki

**Reply by eroboticdude** [reply]
You're saying I will be able to use callbacks? I'll take a look into the wiki.

**Reply by eroboticdude** [reply]
Could you send the link to the wiki page? I am still not able to find detailed use cases for callbacks.

**Reply by pschatzmann** [reply]
https://github.com/pschatzmann/arduino-audio-tools/wiki/Multicore-Processing
